### PR TITLE
Feature: graftable ownership

### PR DESCRIPTION
**Pull request by cap10morgan**
This enables chaintrees to own other chaintrees using the new graftabledag feature in chaintree. Initially for dgit user account recovery use case.

### CODE REVIEW

**Comment by cap10morgan** [comment]
Some things to consider in the review:

1. I don't _love_ setting the `DagGetter` on the `NotaryGroup` but it's currently the only way to get something to a validator function. I tried doing a refactor there but it got pretty gnarly so I abandoned it.
1. The `GetLatest` code for the `DagGetterNode` was semi-cargo-culted from the `client.GetLatest` code. I'm not sure we actually need all of it in there: https://github.com/quorumcontrol/tupelo/blob/16c18b89f5d8ebca188400e9f185a2464fb55e5c/sdk/gossip/types/graftedownership.go#L136
1. Similarly, it seems like we could store the state & checkpoint CIDs on completed rounds even in the memory LRU cache, but the refactor I attempted got a bit messy: https://github.com/quorumcontrol/tupelo/blob/16c18b89f5d8ebca188400e9f185a2464fb55e5c/signer/gossip/node.go#L767

File: sdk/gossip/client/client.go [hunk 1]
```diff
@@ -363,3 +364,7 @@ func (c *Client) SubscribeToAbr(ctx context.Context, abr *services.AddBlockReque
 func (c *Client) UnsubscribeFromAbr(s subscription) {
 	c.subscriber.unsubscribe(s)
 }
+
+func (c *Client) DagGetter() *types.ClientDagGetter {
```
Comment (tobowers) [inline sdk/gossip/client/client.go:368]: these seems like it complects the two types here and that the wrapping should happen at the place that needs to do the wrapping.

File: sdk/gossip/client/client_test.go [hunk 2]
```diff
@@ -120,6 +125,9 @@ func newClient(ctx context.Context, group *types.NotaryGroup, bootAddrs []string
 	if err != nil {
 		return nil, err
 	}
+
+	group.DagGetter = cli.DagGetter()
```
Comment (tobowers) [inline sdk/gossip/client/client_test.go:129]: I actually like this on notary group - I was thinking that in the wasm SDK "Community" should become the NotaryGroup too

File: sdk/gossip/types/graftedownership.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+	newOwners, err := gro.resolveOwnersRecursively(ctx, nextGDag, seen)
+	if err != nil {
+		return nil, err
+	}
+
+	return newOwners, nil
+}
+
+// TODO: Make this detect loops and error?
```
Comment (tobowers) [inline sdk/gossip/types/graftedownership.go:217]: looks like this *does* detect loops now?

File: sdk/gossip/client/client.go (hunk 1 above)
Comment (cap10morgan) [reply sdk/gossip/client/client.go]: removed

File: sdk/gossip/client/client_test.go (hunk 2 above)
Comment (cap10morgan) [reply sdk/gossip/client/client_test.go]: OK, cool. It felt a little weird in the all-nodes-in-the-same-process case, but that's only used in tests and some local dev scenarios.

**Pull request merged**